scipy/optimize/_remove_redundancy.py

```diff
@@ -447,3 +448,70 @@ def _remove_redundancy(A, b):
         s_min = s[-1] if m <= n else 0
 
     return A, b, status, message
+
+
+def _remove_redundancy_id(A, rhs, rank=None, randomized=True):
+    """
+    Eliminates redundant equations from system of equations defined by Ax = b
+    and identifies infeasibilities.
+    Parameters
+    ----------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    rhs : 1-D array
+        An array representing the right-hand side of a system of equations
+    rank : int
+        The rank of A
+    randomized: bool
+        True for randomized interpolative decomposition
+    Returns
+    -------
+    A : 2-D array
+        An array representing the left-hand side of a system of equations
+    rhs : 1-D array
+        An array representing the right-hand side of a system of equations
+    status: int
+        An integer indicating the status of the system
+        0: No infeasibility identified
+        2: Trivially infeasible
+    message : str
+        A string descriptor of the exit status of the optimization.
+    """
+
+    status = 0
+    message = ""
+    inconsistent = ("There is a linear combination of rows of A_eq that "
+                    "results in zero, suggesting a redundant constraint. "
+                    "However the same linear combination of b_eq is "
+                    "nonzero, suggesting that the constraints conflict "
+                    "and the problem is infeasible.")
+
+    A, rhs, status, message = _remove_zero_rows(A, rhs)
+
+    if status != 0:
+        return A, rhs, status, message
+
+    m, n = A.shape
+
+    k = rank
+    if rank is None:
+        k = np.linalg.matrix_rank(A)
+        # can randomized algorithm for ID recognize when it's up to full rank?
+        # it would be nice not to have to calculate/provide the rank
+
+    idx, proj = interp_decomp(A.T, k, rand=randomized)
```

Comment:
One way would be to pass a tolerance value instead. ``interp_decomp`` can interoperate `k` in two possible ways; the required rank or the relative error (i.e. epsilon). The latter occurs if k < 1, in this case the signature becomes: 

``k, idx, proj = interp_decomp(A, eps_or_k)``

which would allow you to use `k` for the slicing later.

Reply:
Thanks, I know about that use. Actually I meant to change that comment - I have to calculate the rank anyway to determine whether redundancy removal should be performed at all - as all of these routines are slower than np.linalg.matrix_rank. If I have it, I might as well use it!